lib/union_of.rb

```diff
@@ -0,0 +1,321 @@
+# frozen_string_literal: true
+
+require 'active_record_union'
+
+module UnionOf
+  class ReadonlyAssociation < ActiveRecord::Associations::CollectionAssociation
+    # TODO(ezekg) Implement a readonly association. Raise proper errors.
+
+    def writer(...)      = raise NotImplementedError
+    def ids_writer       = raise NotImplementedError
+    def destroy_all(...) = raise NotImplementedError
+    def delete_all(...)  = raise NotImplementedError
+
+    def insert_record(...)  = raise NotImplementedError
+    def delete_records(...) = raise NotImplementedError
+    def concat_records(...) = raise NotImplementedError
+    def build_record(...)   = raise NotImplementedError
+  end
+
+  class Association < ReadonlyAssociation
+    def association_scope = Scope.scope(self)
+  end
+
+  class ThroughAssociation < Association
+    include ActiveRecord::Associations::ThroughAssociation
+  end
+
+  class Scope < ActiveRecord::Associations::AssociationScope
+    INSTANCE = create
+
+    def self.scope(association)
+      INSTANCE.scope(association)
+    end
+
+    private
+
+    def last_chain_scope(scope, reflection, owner)
+      return super unless
+        reflection.union_of?
+
+      foreign_table = reflection.aliased_table
```

Comment:
Each union should have an alias derived from the association name, with a `union` suffix.

Reply:
Meh. I'm not sure it's worth the effort.